OSDOCS-11444: Adds low latency to MicroShift (#80236)
openshift/openshift-docs: modules/microshift-low-latency-install-kernelrt-rpm.adoc

// Module included in the following assemblies:
//
// * microshift_configuring/microshift-low-latency.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-low-latency-install-kernelrt_{context}"]
= Installing the {op-system-rt-kernel}

While the real-time kernel is not necessary for low latency workloads, using the {op-system-rtk} can optimize low latency performance. You can install it on a host using RPM packages, and include it in a {op-system-ostree-first} image deployment.

.Prerequisites
* You have a Red Hat subscription that includes {op-system-rt-kernel}. For example, your host machine is registered and Red Hat Enterprise Linux (RHEL) is attached to a RHEL for Real Time subscription.
* You are using x86_64 architecture.

.Procedure
. Enable the {op-system-rtk} repository by running the following command:
+
[source,terminal]
----
$ sudo subscription-manager repos --enable rhel-9-for-x86_64-rt-rpms
----

. Install the real-time kernel by running the following command:
+
[source,terminal]
----

> $ sudo dnf install kernel-rt -y

Reviewer: Might want to keep it as `dnf install -y` as above.

Reviewer 2: agree

----

. Enable the real-time kernel by running the following command:
+
[source,terminal]
----

> $ sudo grubby --set-default="/boot/vmlinuz-$(rpm -q --queryformat '%{version}-%{release}.%{arch}' kernel-rt | sort | tail -1)+rt"

Reviewer 3: I would split this into a simpler command, e.g.
```suggestion
$ RTVER=$(rpm -q --queryformat '%{version}-%{release}.%{arch}' kernel-rt | sort | tail -1)
$ sudo grubby --set-default="/boot/vmlinuz-${RTVER}+rt"
```

// Module included in the following assemblies:
//
// * microshift_configuring/microshift-low-latency.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-low-latency-install-kernelrt_{context}"]
= Installing the {op-system-rt-kernel}

While the real-time kernel is not necessary for low latency workloads, using the {op-system-rtk} can optimize low latency performance. You can install it on a host using RPM packages, and include it in a {op-system-ostree-first} image deployment.

.Prerequisites
* You have a Red Hat subscription that includes {op-system-rt-kernel}. For example, your host machine is registered and Red Hat Enterprise Linux (RHEL) is attached to a RHEL for Real Time subscription.

> //do we need to have installed kernel-rt on the host before we can make the blueprint?

Reviewer 2: No

// Module included in the following assemblies:
//
// * microshift_configuring/microshift-low-latency.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-low-latency-install-kernelrt_{context}"]
= Installing the {op-system-rt-kernel}

> While the real-time kernel is not necessary for low latency workloads, using the {op-system-rtk} can optimize low latency performance. You can install it on a host using RPM packages, and include it in a {op-system-ostree-first} image deployment.

Reviewer 4: Per [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#w).
```suggestion
Although the real-time kernel is not necessary for low latency workloads, using the {op-system-rtk} can optimize low latency performance. You can install it on a host using RPM packages, and include it in a {op-system-ostree-first} image deployment.
```

.Prerequisites
* You have a Red Hat subscription that includes {op-system-rt-kernel}. For example, your host machine is registered and Red Hat Enterprise Linux (RHEL) is attached to a RHEL for Real Time subscription.
* You are using x86_64 architecture.

.Procedure
. Enable the {op-system-rtk} repository by running the following command:
+
[source,terminal]
----
$ sudo subscription-manager repos --enable rhel-9-for-x86_64-rt-rpms
----

. Install the real-time kernel by running the following command:
+
[source,terminal]
----
$ sudo dnf install kernel-rt -y
----

> . Enable the real-time kernel by running the following command:

Reviewer 3: ```suggestion
. Query the real-time kernel version by running the following command:
```

+
[source,terminal]
----
$ RTVER=$(rpm -q --queryformat '%{version}-%{release}.%{arch}' kernel-rt | sort | tail -1)
----
+
. Make a persistent change in GRUB that designates the real-time kernel as the default kernel by running the following command:
+
[source,terminal]
----
$ sudo grubby --set-default="/boot/vmlinuz-${RTVER}+rt"
----

. Restart the host to activate the real-time kernel.

.Next steps
. Prepare your {microshift-short} workloads for low latency.

> . Optional: Use a blueprint to install the real-time kernel in a {op-system-ostree} image.

Reviewer 4: Do we want to include links to make sure folks go to the right places?
